Cargo test fails when multiple versions of `libsmallvec` exist

## Reproduce

```sh
rustup toolchain install nightly --component llvm-tools --component rustc-dev --profile minimal --allow-downgrade --no-self-update
git clone https://github.com/dtolnay/syn
cd syn
cargo test --all-features --release --tests
```

## Expected results

Test passed.

## Actual results

```
   ...
   Compiling reqwest v0.12.26
error[E0464]: multiple candidates for `rmeta` dependency `smallvec` found
  --> tests/test_precedence.rs:40:1
   |
40 | extern crate smallvec;
   | ^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: candidate #1: /var/lib/gitlab-runner/.rustup/toolchains/nightly-loongarch64-unknown-linux-gnu/lib/rustlib/loongarch64-unknown-linux-gnu/lib/libsmallvec-93ff136b46e0054c.rmeta
   = note: candidate #2: /var/lib/gitlab-runner/.rustup/toolchains/nightly-loongarch64-unknown-linux-gnu/lib/rustlib/loongarch64-unknown-linux-gnu/lib/libsmallvec-95d4d31bf50ebc9c.rmeta
For more information about this error, try `rustc --explain E0464`.
error: could not compile `syn` (test "test_precedence") due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

This issue is not specific to loongarch64. The rustc-dev components for armv7, riscv64gc, powerpc, and other targets include multiple `libsmallvec-*.rmeta` files, so they can run into the same issue.

--- a/tests/test_precedence.rs
+++ b/tests/test_precedence.rs
@@ -37,8 +37,6 @@ extern crate rustc_ast_pretty;
 extern crate rustc_data_structures;
 extern crate rustc_driver;
 extern crate rustc_span;
-extern crate smallvec;
-extern crate thin_vec;
 
 use crate::common::eq::SpanlessEq;
 use crate::common::parse;
@@ -210,10 +208,10 @@ fn librustc_parenthesize(mut librustc_expr: Box<ast::Expr>) -> Box<ast::Expr> {
     use rustc_ast::mut_visit::{walk_flat_map_assoc_item, MutVisitor};
     use rustc_ast::visit::{AssocCtxt, BoundKind};
     use rustc_data_structures::flat_map_in_place::FlatMapInPlace;
+    use rustc_data_structures::smallvec::SmallVec;
+    use rustc_data_structures::thin_vec::ThinVec;
     use rustc_span::DUMMY_SP;
-    use smallvec::SmallVec;
     use std::ops::DerefMut;
-    use thin_vec::ThinVec;
 
     struct FullyParenthesize;
 

